src/swtpm/swtpm_io.c

```diff
@@ -205,10 +205,9 @@ TPM_RESULT SWTPM_IO_Init(void)
     if (rc == 0) {
         port_str = getenv("TPM_PORT");
         if (port_str == NULL) {
-            logprintf(STDERR_FILENO,
-                      "SWTPM_IO_Init: Error, TPM_PORT environment variable not "
-                      "set\n");
-            rc = TPM_IOERROR;
+            TPM_DEBUG("SWTPM_IO_Init: TPM_PORT environment variable not set. "
+                      "Peer-to-peer mode enabled.\n");
+            return 0;
```

Comment:
ok, i will rephrase as you suggested.

Reply:
Also the compilation fails. See the Travis test run below.

Reply:
i just had a look at that, i will fix.

Reply:
Another thing is that this fd transfer only works when the control socket is a UnixIO socket but not when it is a TCP socket. It's not a mistake not to give a file descriptor or path in case of UnixIO but in case of TCP it has to be given. I think you currently allow it not to be given in both cases.

Reply:
I think to handle this properly we should move this to parse_server_options(), so that we can exactly  know what kind of sever is configured and limit TPM_PORT environment to "tcp" server. And frankly   SWTPM_IO_ServerSocket_Open() is code duplicate of tcp_open_socket().

This leaves SWTPM_IO_Init() definition to empty. Can you please give your suggestions.

Reply:
It sounds right. Can you make do this in a separate patch?

Reply:
sure, i am working on that.